cdap-cli/src/main/java/co/cask/cdap/cli/CLIConfig.java

```diff
@@ -243,6 +241,13 @@ public InputStream getInput() throws IOException {
     }
   }
 
+  public void setNamespace(Id.Namespace namespace) {
+    ConnectionConfig connectionConfig = ConnectionConfig.builder(clientConfig.getConnectionConfig())
+      .setNamespace(namespace)
+      .build();
+    this.setConnectionConfig(connectionConfig);
```

Comment:
if we're just setting the namespace, do we need to trigger a call to `tryAutoconnect`?